cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/apply/ApplicationRequestsPlaner.java

```diff
@@ -1,164 +1,273 @@
+                            valueChanged.getKey(),
+                            "from value",
+                            valueChanged.getValueBefore(),
+                            "to value",
+                            valueChanged.getValueAfter(),
+                            "for  application",
+                            applicationName);
+                    requests.add(this.appOperations.addEnvironmentVariable(applicationName,
+                            valueChanged.getKey(),
+                            valueChanged.getValueAfter()));
+                } else {
+                    log.debug("Adding request to remove environment variable",
+                            valueChanged.getKey(),
+                            "from application",
+                            applicationName);
+                    requests.add(this.appOperations.removeEnvironmentVariable(applicationName, valueChanged.getKey()));
+                }
+            }
+            return Flux.concat(requests);
+        }
+        return Flux.empty();
+    }
+
+
+    private Publisher<Void> getChangedRoutesRequests(List<CfChange> changes) {
+        Optional<CfChange> optionalRoutesChange = getChange(changes,
+                change -> change.getPropertyName().equals("routes"));
+        List<Mono<Void>> requests = new LinkedList<>();
+
+        if (optionalRoutesChange.isPresent()) {
+            CfContainerChange routesChanges = (CfContainerChange) optionalRoutesChange.get();
+            logChange(routesChanges);
+
+            for (CfContainerValueChanged valueChanged : routesChanges.getValueChangesBy(ChangeType.ADDED)) {
+                log.debug("Adding request to add route",
+                        valueChanged.getValue(),
+                        "to application",
+                        applicationName);
+                requests.add(this.appOperations.addRoute(applicationName, valueChanged.getValue()));
+            }
+
+            for (CfContainerValueChanged valueChanged : routesChanges.getValueChangesBy(ChangeType.REMOVED)) {
+                log.debug("Adding request to remove route",
+                        valueChanged.getValue(),
+                        "from application",
+                        applicationName);
+                requests.add(this.appOperations.removeRoute(applicationName, valueChanged.getValue()));
+            }
+
         }
+        return Flux.concat(requests);
+    }
+
+
+    private Mono<Void> getScaleInstancesRequest(List<CfChange> changes) {
+        Optional<CfChange> instancesChange = getChange(changes,
+                change -> change.getPropertyName().equals("instances"));
+
+        if (instancesChange.isPresent()) {
+            logChange(instancesChange.get());
+
+            ApplicationBean bean = (ApplicationBean) instancesChange.get().getAffectedObject();
+            return this.appOperations.scale(applicationName,
+                    null,
+                    null,
```

Comment:
A comment could explain, that a restart is required for these properties (that's why they are null)